### ISSUE TITLE
Game scale issue with volume buttons

### ISSUE THREAD

**Issue by Zebra1337** [OpenRCT2/OpenRCT2#4394]
**OS:** Ubuntu 
**Version:** 16.04.1
**Commit/Build:** d00aa2c

When using the volume keyboard buttons on my laptop, the game kinda zooms / scales in whilst the volume notification appears behind it. I imagine this will happen for all notifications. Happens when using a smaller resolution than native. 

I run the game at 1280x720 on a 1920x1080 screen.

**Steps to reproduce:**
1. Run OpenRCT2 on a smaller resolution than your desktop resolution
2. Trigger an ubuntu notification such as changing volume. 

**Screenshots / Video:**
This is it when the notification is triggered: http://i.imgur.com/tulXfz6.png
It goes back to normal after it disappears.

**Save game:**
Standard Six Flags Magic Mountain, any.

**Comment by janisozaur** [comment]
`latest` is not a valid commit.

**Comment by janisozaur** [comment]
I don't understand what this issue is really about. Can you explain it further?

**Comment by Zebra1337** [comment]
Okay, you know when a notification appears on Ubuntu? Such as using the volume hotkeys on a keyboard you get this: http://i.imgur.com/e1wRKKU.png

Well that appears under the game, and at the same time causes the game to sorta zoom / scale. I imagine what's happening is the window is being stretched to my desktop resolution, 1920x1080.

Notifications fade after a few seconds, and once that happens it goes back to normal. 

It could well be an Ubuntu problem as it seems to be even worse running RCT2 on Wine:
http://i.imgur.com/rOBEEub.png

This sort of game I feel better playing at a lower resolution so it's a shame.

**Comment by janisozaur** [comment]
It sounds like the problem lies with Ubuntu's builds of SDL2 or Unity. Can you detail what display options you use or provide your `config.ini`?

**Comment by AlexRainbows** [comment]
More than likely Unity I imagine, grr. Can't find nothing of it online. It works fine if I change my desktop resolution to 1280x720 but changing that back and forth each time will get quite annoying.

`[general]
always_show_gridlines = false
autosave = 1
confirmation_prompt = false
construction_marker_colour = 0
currency_format = GBP
custom_currency_rate = 10
custom_currency_affix = SUFFIX
custom_currency_symbol = "Ctm"
edge_scrolling = true
fullscreen_mode = 1
fullscreen_height = 720
fullscreen_width = 1280
game_path = "/home/alex/PlayOnLinux's virtual drives/RollerCoasterTycoon2_gog/drive_c/GOG Games/RollerCoaster Tycoon 2 Triple Thrill Pack"
landscape_smoothing = true
language = en-GB
measurement_format = METRIC
play_intro = false
save_plugin_data = false
debugging_tools = false
show_height_as_units = false
temperature_format = CELSIUS
window_height = 480
window_snap_proximity = 5
window_width = 720
drawing_engine = SOFTWARE
uncap_fps = false
test_unfinished_tracks = false
no_test_crashes = false
date_format = DD/MM/YY
auto_staff = false
handymen_mow_default = false
default_inspection_interval = 2
last_run_version = "0.0.5"
invert_viewport_drag = false
load_save_sort = 0
minimize_fullscreen_focus_loss = true
day_night_cycle = false
upper_case_banners = false
disable_lightning_effect = false
allow_loading_with_incorrect_checksum = false
steam_overlay_pause = true
window_scale = 1.000
scale_quality = 2
use_nn_at_integer_scales = true
show_fps = false
trap_cursor = false
auto_open_shops = false
scenario_select_mode = 1
scenario_unlocking_enabled = true
scenario_hide_mega_park = true
last_game_directory = ""
last_landscape_directory = ""
last_scenario_directory = ""
last_track_directory = ""
window_limit = 64
zoom_to_cursor = true

[interface]
toolbar_show_finances = false
toolbar_show_research = true
toolbar_show_cheats = false
toolbar_show_news = false
select_by_track_type = false
console_small_font = false
current_theme = "RCT2"
current_title_sequence = "*OPENRCT2"
object_selection_filter_flags = 2031

[sound]
master_volume = 100
title_music = 2
sound = true
sound_volume = 100
ride_music = true
ride_music_volume = 100
audio_focus = false
audio_device = ""

[twitch]
channel = ""
follower_peep_names = true
follower_peep_tracking = false
chat_peep_names = true
chat_peep_tracking = true
news = false

[network]
player_name = "alex"
default_port = 11753
default_password = ""
stay_connected = true
advertise = false
maxplayers = 16
server_name = "Server"
server_description = ""
server_greeting = ""
master_server_url = ""
provider_name = ""
provider_email = ""
provider_website = ""
known_keys_only = false
log_chat = false

[notifications]
park_award = true
park_marketing_campaign_finished = true
park_warnings = true
park_rating_warnings = true
ride_broken_down = true
ride_crashed = true
ride_warnings = true
ride_researched = true
guest_warnings = true
guest_lost = false
guest_entered_left_park = true
guest_queuing_for_ride = true
guest_on_ride = true
guest_left_ride = true
guest_bought_item = true
guest_used_facility = true
guest_died = true

[fonts]
file_name = ""
font_name = ""
x_offset = 0
y_offset = -1
size_tiny = 8
size_small = 10
size_medium = 11
size_big = 12
height_tiny = 6
height_small = 12
height_medium = 12
height_big = 20
`

**Comment by janisozaur** [comment]
@AlexRainbows are you @Zebra1337 ?

I don't have a way of testing it, as I can only run fullscreen in my native resolution. I can only suggest you use your native resolution as well with in-game `window scale` option.

@Gymnasiast can you test that?

**Comment by Zebra1337** [comment]
I am, sorry :p I made accounts ages ago. One posted from my notebook.

The scaling option goes pixelated so I don't really like that. 

It runs without faults on my Fedora 24 desktop so likely a unity issue.

**Comment by IntelOrca** [comment]
Have you tried switching to hardware display graphics mode?

**Comment by Zebra1337** [comment]
Just tried that but it didn't solve it. A video I just recorded shows whats happening, let me upload it.

**Comment by Zebra1337** [comment]
Here is the video: https://vid.me/Ocyj

The recording actually reveals what is happening better. When I'm in-game I only see the top-left at 1280x720 and I don't see the notification. When the notification appears on the video, at the same time it stretches the window to 1080, but in-game I only see the top left of it stretched out.

Weird bug.

**Comment by marijnvdwerf** [comment]
Can you test setting the scale at 1.5 and selecting your actual resolution in-game?

**Comment by Zebra1337** [comment]
Tried that but the text and things become blocky. This is such an annoying problem. I think something relating to fullscreen applications is broken in Ubuntu 16.04. 

Fedora on my desktop has this working seamlessly.

**Comment by janisozaur** [comment]
it sounds like you only tested nearest neighbour, have you tried linear scaling as well?

**Comment by Zebra1337** [comment]
Interestingly Linear was highlighted but it was actually using nearest neighbour. Linear & Anisotropic look 10x better and how I wanted it. The game is now perfectly playable, thank you!

The original bug definitely seems to be a problem with Unity, as I tested it earlier on GNOME and it worked fine, so I will file a bug report on this.

**Comment by janisozaur** [comment]
I will close this now, based on your latest comment.
